File: backend/models/comment.py
import uuid
from sqlalchemy import Column, Text, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from database import Base


class Comment(Base):
    __tablename__ = "comments"

    comment_id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    content = Column(Text, nullable=False)

    card_id = Column(UUID(as_uuid=True), ForeignKey("cards.card_id"), nullable=False)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.user_id"), nullable=False)

    card = relationship("Card", back_populates="comments")
    user = relationship("User", back_populates="comments")

File: frontend/src/types/comment.ts
export interface CommentUser {
  user_id: string;
  name: string;
  email: string;
}

export interface Comment {
  comment_id: string;
  content: string;
  card_id: string;
  user_id: string;
  created_at: string;
  user: CommentUser;
}

export interface CreateCommentPayload {
  content: string;
}

export interface CommentsResponse {
  message: string;
  data: Comment[];
}

export interface CommentResponse {
  message: string;
  data: Comment;
}
